From e1a09b4b20b1fefdc64228d798bf45aead1887c5 Mon Sep 17 00:00:00 2001
From: Kramer Ge <fangzhoug@chromium.org>
Date: Mon, 29 Sep 2025 13:54:54 -0700
Subject: [PATCH] [Ozone/Wayland]Remove DCHECK superceded by preferred_scale

This DCHECK assumes the widget tree should have the same window_scale
because they're on the same display and hence should deduce the same
scale.

This assumption is not true when preferred_scale protocol is used, where
compositor sends each widget scale factor asynchronously.

Bug: 446602690
Change-Id: I35432320b87f337651dc466afb69d5984b5f69bb
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/6997107
Commit-Queue: Kramer Ge <fangzhoug@chromium.org>
Reviewed-by: Thomas Anderson <thomasanderson@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1522463}
---
 ui/ozone/platform/wayland/common/wayland_util.cc | 2 --
 1 file changed, 2 deletions(-)

diff --git a/ui/ozone/platform/wayland/common/wayland_util.cc b/ui/ozone/platform/wayland/common/wayland_util.cc
index 2da49f4042345..749013e5445d5 100644
--- a/ui/ozone/platform/wayland/common/wayland_util.cc
+++ b/ui/ozone/platform/wayland/common/wayland_util.cc
@@ -284,8 +284,6 @@ gfx::Rect TranslateWindowBoundsToParentDIP(ui::WaylandWindow* window,
                                            ui::WaylandWindow* parent_window) {
   DCHECK(window);
   DCHECK(parent_window);
-  DCHECK_EQ(window->applied_state().window_scale,
-            parent_window->applied_state().window_scale);
   return wl::TranslateBoundsToParentCoordinates(
       window->GetBoundsInDIP(), parent_window->GetBoundsInDIP());
 }
-- 
2.51.0

